util/types/json/impl.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package json
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/juju/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+var _ JSON = &jsonImpl{}
+
+type jsonImpl struct {
+	json interface{}
+}
+
+func (j *jsonImpl) ParseFromString(s string) (err error) {
+	if len(s) == 0 {
+		return ErrInvalidJSONText.GenByArgs("The document is empty")
+	}
+	err = bson.UnmarshalJSON([]byte(s), &j.json)
+	if err != nil {
+		return ErrInvalidJSONText.GenByArgs(err)
+	}
+	return err
+}
+
+func (j *jsonImpl) DumpToString() string {
```

Comment:
What's the different between DumpToString and Serialize ? Why we return DumpToString to MySQL Client?